src/main/java/uk/gov/hmcts/darts/event/service/impl/DefaultEventHandler.java

```diff
@@ -0,0 +1,102 @@
+import uk.gov.hmcts.darts.event.model.DartsEvent;
+import uk.gov.hmcts.darts.event.service.DarNotifyService;
+
+import java.time.OffsetDateTime;
+
+import static java.lang.String.format;
+import static java.lang.String.join;
+import static java.util.Objects.isNull;
+import static uk.gov.hmcts.darts.event.enums.DarNotifyType.CASE_UPDATE;
+
+@Slf4j
+@Service
+@RequiredArgsConstructor
+@SuppressWarnings("PMD.AvoidLiteralsInIfCondition")
+public class DefaultEventHandler extends EventHandlerBase {
+
+    private static final String NOT_FOUND_MESSAGE = "Courthouse: %s & Courtroom: %s combination not found in database";
+    private static final String MULTIPLE_CASE_NUMBERS = "Event: %s contains multiple caseNumbers: %s";
+    private final CaseRepository caseRepository;
+    private final EventRepository eventRepository;
+    private final CourtroomRepository courtroomRepository;
+    private final HearingRepository hearingRepository;
+    private final DarNotifyService darNotifyService;
+
+    @Override
+    @Transactional
+    public void handle(final DartsEvent dartsEvent) {
+        final var actualCourtRoomEntity = courtroomRepository
+            .findByCourthouseNameAndCourtroomName(dartsEvent.getCourthouse(), dartsEvent.getCourtroom())
+            .orElseThrow(() -> new DartsApiException(
+                  EventError.EVENT_DATA_NOT_FOUND,
+                  format(NOT_FOUND_MESSAGE, dartsEvent.getCourthouse(), dartsEvent.getCourtroom())));
+
+        final var caseNumbers = dartsEvent.getCaseNumbers();
+        if (caseNumbers.size() > 1) {
+            log.info(format(MULTIPLE_CASE_NUMBERS, dartsEvent.getEventId(), join(", ", caseNumbers)));
+        }
+        final var actualCourtHouse = actualCourtRoomEntity.getCourthouse();
+        final var actualEventCaseNumber = caseNumbers.get(0);
+        final var actualEventDate = dartsEvent.getDateTime();
+
+        var actualCourtCaseEntity = caseRepository
+            .findByCaseNumberAndCourthouse_CourthouseName(actualEventCaseNumber, actualCourtHouse.getCourthouseName())
+            .orElseGet(() -> createNewCaseAt(actualCourtHouse, actualEventCaseNumber));
+
+        var actualHearingEntity = actualCourtCaseEntity.getHearings().stream()
+            .filter(hearingEntity -> hearingEntity.isFor(actualEventDate))
+            .findFirst().orElseGet(() -> newCaseHearing(actualCourtRoomEntity, actualEventDate, actualCourtCaseEntity));
+
+        actualHearingEntity.setHearingIsActual(true);
+
+        var eventEntity = eventEntityFrom(dartsEvent);
+        eventEntity.setCourtroom(actualCourtRoomEntity);
+        eventRepository.save(eventEntity);
+
+        if (eitherTheHearingIsNewOrTheCourtroomIsDifferent(actualCourtRoomEntity, actualHearingEntity)) {
+            actualHearingEntity.setCourtroom(actualCourtRoomEntity);
+            darNotifyService.darNotify(dartsEvent, CASE_UPDATE);
+        }
+
+        caseRepository.save(actualCourtCaseEntity);
```

Comment:
Why do we need to call `save` on `Case` - did we change something?

Reply:
Because we might have created if it wasn't in the database.  Its hard to follow I know.